metaflow/plugins/kfp/kfp.py

```diff
@@ -181,11 +182,14 @@ def __init__(
         self.notify_on_success = notify_on_success
         self._client = None
 
-    def create_run_on_kfp(self, run_name: str, flow_parameters: dict):
-        """
-        Creates a new run on KFP using the `kfp.Client()`.
+    def get_kubernetes_service_account(self) -> str:
+        """Returns the service account from an individual profile notebook if a user
+        inputs an IAM role for their notebook. If no ServiceAccount is set by the user,
+        default to the`deafult-editor` ServiceAccount.
         """
-        # TODO: first create KFP Pipeline, then an experiment if provided else default experiment.
+        return KUBERNETES_SERVICE_ACCOUNT if not None else "default-editor"
```

Comment:
Can we simplify the client-side implementation here by just not defaulting here.

Basically if we don't set `service_account` in `create_run_from_pipeline_func` I presume it defaults to `pipeline-runner` which then delegates the "default" `serviceAccount` name to the KFP API service I think?

If we do the same down on line 233 by doing something like:

```python
    if KUBERNETES_SERVICE_ACCOUNT:
        self.set_kfp_client()
        workflow_yaml = self._client._extract_pipeline_yaml(pipeline_file_path)
        workflow_yaml["spec"][
            "serviceAccountName"
        ] = self.get_kubernetes_service_account()
        kfp.compiler.Compiler()._write_workflow(workflow_yaml, pipeline_file_path)
```

Then we can totally avoid needing to hard code the default value in the SDK and meaning if for whatever reason we want to change that we can do it without pushing out a client upgrade unless I've missed something?

Reply:
yep. made the changes